src/Illuminate/Auth/DatabaseUserProvider.php

```diff
@@ -110,7 +111,17 @@ public function retrieveByCredentials(array $credentials)
         $query = $this->conn->table($this->table);
 
         foreach ($credentials as $key => $value) {
-            if (! Str::contains($key, 'password')) {
+            if (Str::contains($key, 'password')) {
+                continue;
+            }
+
+            if ($value instanceof Arrayable || $value instanceof Collection) {
+                $value = $value->toArray();
```

Comment:
I believe that `whereIn` now supports `Arrayable`/`Collection` arguments so there shouldn't bee a need to manually cast it to an array here. I suspect you could do something like this - if you think it's an improvement.

```php
if (is_array($value) || $value instanceof Arrayable || $value instanceof Collection) {
    $query->whereIn($key, $value);
} else {
    $query->where($key, $value);
}
```

Reply:
Why checking for `$value instanceof Collection` ? `Illuminate\Support\Collection` already implements `Arrayable `

Reply:
Also @VinceG , make sure to import `Arrayable`:
`use Illuminate\Contracts\Support\Arrayable;`